TestingHelpers/MockFileData.cs

```diff
@@ -97,11 +98,47 @@ public MockFileData(byte[] contents)
         }
 
         /// <summary>
-        /// Gets or sets the byte contents of the <see cref="MockFileData"/>.
+        /// Allows insertion of a file where the content is lazy loaded through a Func.
         /// </summary>
+        /// <param name="lazyLoadContent">Func to return the file's content the first time it is requested.</param>
+        public MockFileData(Func<byte[]> lazyLoadContent)
+        {
+            if (lazyLoadContent == null)
+            {
+                throw new ArgumentNullException("lazyLoadContent");
+            }
+            this.lazyLoadContent = lazyLoadContent;
+        }
+
+        /// <summary>
+        /// Allows insertion of a FileSystemInfoBase object into a new file system. Attributes and time stamps
+        /// are copied, and the content will be lazy loaded the first time it is touched.
+        /// </summary>
+        /// <param name="fileSystem">The source file system where the file info originates</param>
+        /// <param name="fileInfo">The source file to be copied</param>
+        public MockFileData(IFileSystem fileSystem, FileSystemInfoBase fileInfo)
```

Comment:
Well, it's closing in on three years since I wrote this, and I realize I have forgotten all about it. :-( 1) What do you propose we should do instead? 2) would it be easier to entirely drop this PR and that I create a new one from the current master, with your suggested changes?